src/TestFramework/TestFramework/TestDataRow.cs

```diff
@@ -40,6 +40,12 @@ public TestDataRow(T value)
     [DataMember]
     public string? DisplayName { get; set; }
 
+    /// <summary>
+    /// Gets or sets the test categories for the test case.
+    /// </summary>
+    [DataMember]
+    public IList<string>? TestCategories { get; set; }
```

Comment:
@Youssef1313 what about using a readonly collection here? We don't really expect to modify the collection from our end. The benefit of keeping it this way is to allow users to create the `TestDataRow` instance and add categories after but the logic could be reverted.

Reply:
@Evangelink No strong opinion from me here. So I'll go with whichever you prefer :)